WooCommerce/Classes/ViewRelated/Orders/Order Details/Product List Section/Product Details/AggregatedProductListViewController.swift

```diff
@@ -0,0 +1,148 @@
+    private var products: [Product]? = []
+    var items: [AggregateOrderItem] = []
+
+    private let imageService: ImageService = ServiceLocator.imageService
+
+    // MARK: - View Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.products = viewModel.products
+
+        configureMainView()
+        configureTableView()
+    }
+}
+
+
+// MARK: - Configuration
+//
+private extension AggregatedProductListViewController {
+
+    /// Setup: Main View
+    ///
+    func configureMainView() {
+        title = String.pluralize(items.count, singular: Localization.titleSingular, plural: Localization.titlePlural)
+        view.backgroundColor = .listBackground
+    }
+
+    /// Setup: TableView
+    ///
+    func configureTableView() {
+        tableView.backgroundColor = .listBackground
+        tableView.estimatedRowHeight = Constants.rowHeight
+        tableView.rowHeight = UITableView.automaticDimension
+        tableView.registerNib(for: PickListTableViewCell.self)
+    }
+}
+
+
+// MARK: - UITableViewDataSource Conformance
+//
+extension AggregatedProductListViewController: UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return items.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let item = itemAtIndexPath(indexPath)
+        let product = lookUpProduct(by: item.productOrVariationID)
+        let addOns = itemAddOnsAttributes(item: item)
+        let itemViewModel = ProductDetailsCellViewModel(aggregateItem: item,
+                                                        currency: viewModel.order.currency,
+                                                        product: product,
+                                                        hasAddOns: addOns.isNotEmpty)
+        let cell = tableView.dequeueReusableCell(PickListTableViewCell.self, for: indexPath)
+        cell.selectionStyle = .default
+        cell.configure(item: itemViewModel, imageService: imageService)
+        cell.onViewAddOnsTouchUp = { [weak self] in
+            self?.itemAddOnsButtonTapped(addOns: addOns)
+        }
+
+        return cell
+    }
+}
+
+
+// MARK: - UITableViewDelegate Conformance
+//
+extension AggregatedProductListViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+
+        let orderItem = itemAtIndexPath(indexPath)
+        productWasPressed(orderItem: orderItem)
+    }
+}
+
+
+// MARK: - Private Helpers
+//
+private extension AggregatedProductListViewController {
+
+    func itemAtIndexPath(_ indexPath: IndexPath) -> AggregateOrderItem {
+        return items[indexPath.row]
+    }
+
+    func lookUpProduct(by productID: Int64) -> Product? {
+        return products?.filter({ $0.productID == productID }).first
+    }
```

Comment:
I think this logic should be inside the view model. That would allow us to add some tests to it.